src/Compilers/CSharp/Portable/Symbols/Source/SourceParameterSymbol.cs

```diff
@@ -71,7 +71,7 @@ internal abstract class SourceParameterSymbol : SourceParameterSymbolBase
                 (syntax.AttributeLists.Count == 0) &&
                 !owner.IsPartialMethod())
             {
-                return new SourceSimpleParameterSymbol(owner, parameterType, ordinal, refKind, name, locations);
+                return new SourceSimpleParameterSymbol(owner, parameterType, ordinal, refKind, name, isDiscard: false, locations);
```

Comment:
>return new SourceSimpleParameterSymbol(owner, parameterType, ordinal, refKind, name, isDiscard: false, locations); [](start = 16, length = 114)

Are we making an assumption that this function is never called to create a lambda parameter? Consider adding an assert at the beginning of the method that ```owner``` is not a ```LambdaSymbol```.